code/game/atoms.dm

```diff
@@ -1121,8 +1128,20 @@ GLOBAL_LIST_EMPTY(blood_splatter_icons)
 /atom/proc/zap_act(power, zap_flags)
 	return
 
-/atom/proc/handle_ricochet(obj/item/projectile/P)
-	return
+/atom/proc/handle_ricochet(obj/item/projectile/ricocheting_projectile)
+	var/turf/p_turf = get_turf(ricocheting_projectile)
+	var/face_direction = get_dir(src, p_turf) || get_dir(src, ricocheting_projectile)
```

Comment:
is this supposed to be `||` or `|` ?

Reply:
The `||` means that `face_direction` will get the value of the first `get_dir` call that doesn't return 0. This is what we want in case `src` shares the same `loc` as either the projectile's turf or the projectile itself.